src/DocumentDB.ChangeFeedProcessor/LeaseManagement/DocumentServiceLeaseStoreManager.cs

```diff
@@ -107,6 +110,9 @@ public async Task<ILease> CreateLeaseIfNotExistAsync(string partitionId, string
                 ContinuationToken = continuationToken,
             };
 
+            if (this.isPartitionedByLeasePk)
```

Comment:
we should just call request options factory to AddParititonKeyIfNeeded. Not special cases, if-branches at this layer here.